libevmcore/Assembly.cpp

```diff
@@ -156,6 +163,133 @@ ostream& Assembly::stream(ostream& _out, string const& _prefix, StringMap const&
 	return _out;
 }
 
+Json::Value Assembly::createJsonValue(string _name, int _locationX, int _locationY, string _value, string _jumpType) const
+{
+	Json::Value value;
+	value["name"] = _name;
+	value["begin"] = _locationX;
+	value["end"] = _locationY;
+	if (!_value.empty())
+		value["value"] = _value;
+	if (!_jumpType.empty())
+		value["jumpType"] = _jumpType;
+	return value;
+}
+
+Json::Value Assembly::streamAsmJson(ostream& _out, string const& _prefix, StringMap const& _sourceCodes, bool _inJsonFormat) const
+{
+	Json::Value root;
+
+	string currentArrayName = ".code";
+	std::vector<Json::Value> currentCollection;
+	for (AssemblyItem const& i: m_items)
+	{
+		switch (i.type())
+		{
+		case Operation:
+			currentCollection.push_back(
+				createJsonValue(instructionInfo(i.instruction()).name, i.getLocation().start, i.getLocation().end, i.getJumpTypeAsString()));
+			break;
+		case Push:
+		{
+			std::stringstream hexStr;
+			hexStr << hex << i.data();
+			currentCollection.push_back(
+				createJsonValue(string("PUSH"), i.getLocation().start, i.getLocation().end, hexStr.str(), i.getJumpTypeAsString()));
+		}
+			break;
+		case PushString:
+			currentCollection.push_back(
+				createJsonValue(string("PUSH tag"), i.getLocation().start, i.getLocation().end, m_strings.at((h256)i.data())));
+			break;
+		case PushTag:
+			currentCollection.push_back(
+				createJsonValue(string("PUSH [tag]"), i.getLocation().start, i.getLocation().end, string(i.data())));
+			break;
+		case PushSub:
+			currentCollection.push_back(
+				createJsonValue(string("PUSH"), i.getLocation().start, i.getLocation().end, string("[$]" + dev::toString(h256(i.data())))));
+			break;
+		case PushSubSize:
+			currentCollection.push_back(
+				createJsonValue(string("PUSH"), i.getLocation().start, i.getLocation().end, string("#[$]" + dev::toString(h256(i.data())))));
+			break;
+		case PushProgramSize:
+			currentCollection.push_back(
+				createJsonValue(string("PUSHSIZE"), i.getLocation().start, i.getLocation().end));
+			break;
+		case Tag:
+		{
+			Json::Value collection(Json::arrayValue);
+			for (auto it: currentCollection)
+				collection.append(it);
+			currentCollection.clear();
+			root[currentArrayName] = collection;
+			currentArrayName = "tag" + string(i.data());
```

Comment:
`i.data()` should be a value field of the json, not just concatenated to form the name of the tag.